opentracing/tracer.py

```diff
@@ -34,44 +33,53 @@ class Tracer(SpanPropagator):
 
     singleton_noop_span = Span()
 
-    def start_trace(self, operation_name=None, tags=None):
-        """Starts a new trace and creates a new root span.
+    def start_span(self, operation_name, parent=None, **kwargs):
+        """Starts and returns a new Span representing a unit of work.
 
-        This method should be used by services that are instrumented for
-        tracing but did not receive trace ID from upstream request.
+        :param operation_name: name of the operation represented by the new
+            span from the perspective of the current service.
+        :param parent: an optional parent Span. If specified, the returned Span
+            will be a child of `parent` in `parent`'s trace. If unspecified,
+            the returned Span will be the root of its own trace.
+        :param tags: optional dictionary of Span Tags. The caller gives up
+            ownership of that dictionary, because the Tracer may use it as is
+            to avoid extra data copying.
+        :param start_time: an explicit Span start timestamp as a unix timestamp
+            per time.time()
 
-        :param operation_name: the service's own name for the end-point
-            that received the request represented by this trace and span.
-            The domain of names must be limited, e.g. do not use UUIDs or
-            entity IDs or timestamps as part of the name.
-        :param tags: optional dictionary of Span Tags. The caller is
-            expected to give up ownership of that dictionary, because the
-            Tracer may use it as is to avoid extra data copying.
-
-        :return: a new root Span
+        :return: Returns a new child Span in "started" state.
         """
         return Tracer.singleton_noop_span
 
-    def join_trace(self, operation_name, parent_span, tags=None):
-        """Joins a trace started elsewhere and creates a new span as a
-        child of the given parent trace context.
+    def injector(self, format):
```

Comment:
I'm not a python expert, but doesn't look like it has sensible enums, so yea I would go for strings, or maybe opaque objects, as static members of a class:
```python
class Format(object):
    SPLIT_TEXT = object()

# elsewhere
def injector(self, format):
    if format is Format.SPLIT_TEXT:
        # do it
```